searchlib/src/vespa/searchlib/features/debug_attribute_wait.cpp

```diff
@@ -28,39 +25,26 @@ class DebugAttributeWaitExecutor : public FeatureExecutor
     void execute(uint32_t docId) override;
 };
 
-DebugAttributeWaitExecutor::DebugAttributeWaitExecutor(const IQueryEnvironment &env,
+DebugAttributeWaitExecutor::DebugAttributeWaitExecutor(const IQueryEnvironment &,
                                                        const IAttributeVector *attribute,
                                                        const DebugAttributeWaitParams &params)
     : _attribute(attribute),
       _buf(),
       _params(params)
-{
-    (void)env;
-}
+{ }
 
 void
 DebugAttributeWaitExecutor::execute(uint32_t docId)
 {
     double waitTime = 0.0;
-    FastOS_Time time;
-    time.SetNow();
 
-    if (_attribute != NULL) {
+    if (_attribute != nullptr) {
         _buf.fill(*_attribute, docId);
         waitTime = _buf[0];
     }
-    double millis = waitTime * 1000.0;
-
-    while (time.MilliSecsToNow() < millis) {
-        if (_params.busyWait) {
-            for (int i = 0; i < 1000; i++)
-                ;
-        } else {
-            int rem = (int)(millis - time.MilliSecsToNow());
-            std::this_thread::sleep_for(std::chrono::milliseconds(rem));
-        }
-    }
-    outputs().set_number(0, 1.0e-6 * time.MicroSecsToNow());
+    fastos::StopWatch timer;
+    fastos::StopWatch::waitAtLeast(microseconds(static_cast<long>(waitTime * 1000000)), _params.busyWait);
```

Comment:
I prefer long(expr) to static cast in these cases

Reply:
Fixed